overlord/devicestate/firstboot_preseed_test.go

```diff
@@ -191,6 +187,47 @@ func checkPreseedOrder(c *C, tsAll []*state.TaskSet, snaps ...string) {
 			c.Check(waitTasks[0], Equals, prevTask)
 		}
 
+		// make sure that install-hooks wait for the previous snap, and for
+		// mark-preseeded.
+		hookEdgeTask, err := ts.Edge(snapstate.HooksEdge)
+		c.Assert(err, IsNil)
+		c.Assert(hookEdgeTask.Kind(), Equals, "run-hook")
+		var hsup hookstate.HookSetup
+		c.Assert(hookEdgeTask.Get("hook-setup", &hsup), IsNil)
+		c.Check(hsup.Hook, Equals, "install")
+		if hsup.Snap != "core" && hsup.Snap != "core18" && hsup.Snap != "snapd" {
+			var waitsForMarkPreseeded, waitsForPreviousSnapHook, waitsForPreviousSnap bool
+			for _, wt := range hookEdgeTask.WaitTasks() {
+				switch wt.Kind() {
+				case "setup-aliases":
+					continue
+				case "run-hook":
+					var hsup hookstate.HookSetup
```

Comment:
maybe name this var differently just to be double sure we aren't shadowing the `hsup` outside the loop?